dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -747,6 +749,42 @@ public User updateLoggedInUserMetadata(@ApiParam(hidden = true)@Parameter(hidden
         return dbuser;
     }
 
+    @PUT
+    @Timed
+    @UnitOfWork
+    @RolesAllowed({"admin", "curator"})
+    @Path("/user/{userId}/privileges")
+    @Consumes("application/json")
+    @Operation(operationId = "setUserPrivileges", description = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiOperation(value = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = User.class, hidden = true)
+    public User setUserPrivilege(@Parameter(hidden = true, name = "user")@Auth User authUser,
+                                 @Parameter(name = "User ID", required = true) @PathParam("userId") Long userID,
+                                 @Parameter(name = "Set privilege for a user", required = true) PrivilegeRequest privilegeRequest) {
+        User user = userDAO.findById(userID);
+        if (user == null) {
+            throw new CustomWebApplicationException("User not found", HttpStatus.SC_NOT_FOUND);
+        }
+
+        //This ensures that the user cannot modify their own privileges.
+        if (authUser.getId() == user.getId()) {
+            throw new CustomWebApplicationException("You cannot modify your own privileges", HttpStatus.SC_FORBIDDEN);
+        }
+
+        //If the request's admin setting is different than the admin status of the user that is being modified, and the auth user is not an admin: Throw an error.
+        //This ensures that a curator cannot modify the admin status of any user.
+        if (privilegeRequest.isAdmin() != user.getIsAdmin() && !authUser.getIsAdmin()) {
+            throw new CustomWebApplicationException("You do not have privileges to modify administrative rights", HttpStatus.SC_FORBIDDEN);
+        }
+
+        //Else if the request's settings is different from the privileges of the user that is being modified: update the privileges with the request
```

Comment:
This real nit-picky, and I'm surprised the checkstyle isn't flagging it already (maybe it only catches it in TypeScript, because I'm pretty sure I've seen it in my code), but you can put a space before "Else" --> `// Else...`

Reply:
Done.